### PR TITLE
`Development`: Extend cypress e2e documentation

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Currently setting up the E2E tests locally can be challenging.

### Description

 This PR adds more documentation about how to setup cypress locally and also adds best practices on how to write new tests.

#### Code Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

**Review by TimOrtel — APPROVED** [review]
Thank you for this. I think this is a very much needed extension of our documentation. 

Is it possible to run the Cypress tests from a clean state without any external dependencies? For example if I want to configure them to use LocalCi/Vc and a local user database, instead of Jira. 

If this is possible I think it would also fit quite nicely into this documentation.

**Comment by TheZoker** [comment]
Well actually the E2E tests don't really care, if you use local VC/CI, bitbucket/bamboo or gitlab/jenkins. You only need to setup Artemis to work with those services and we already have documentation covering that. The rest of the setup is similar (just add the users in Artemis and set the user ID in the cypress config). 

By default it is not possible to run the cypress tests without any configuration and without any external dependencies, since Artemis always need a CI/VC service for the programming exercises, but that's independent of cypress. BTW if you use `npm run cypress:open` you can run the tests selectively (as written in the new documentation) and only run tests for certain parts (e.g. skip the programming exercises because of missing CI/VC setup).

File: docs/dev/cypress.rst [hunk 1]
```diff
@@ -18,6 +18,155 @@ the external services (Jira, Bitbucket and Bamboo) and connect them directly wit
...
+
+  You can now click on any test suite and it should run. 
+  **IMPORTANT**: If you run the E2E tests for the first time, always run the ``ImportUsers.ts`` tests first,
+  since it will create the necessary users.
+
+
+Debug using Sorry Cypress
+-------------------------
+In order to debug failing E2E Cypress tests with a dashboard (called Sorry Cypress), please follow the instructions
+on confluence: https://confluence.ase.in.tum.de/display/ArTEMiS/Sorry+Cypress+Dashboard
```
Comment (pal03377) [inline docs/dev/cypress.rst:98]: Shouldn't we put this into the documentation directly (except for the credentials of course)?

File: docs/dev/cypress.rst [hunk 2]
```diff
@@ -18,6 +18,155 @@ the external services (Jira, Bitbucket and Bamboo) and connect them directly wit
...
+
+
+Best Practice when writing new E2E tests
```
Comment (pal03377) [inline docs/dev/cypress.rst:101]:
```suggestion
Best practice when writing new E2E tests
```

consistency

**Review by pal03377 — APPROVED** [review]
Thanks for the documentation improvement! I only have two small non-crutial suggestions.

File: docs/dev/cypress.rst (hunk 1 above)
Comment (TheZoker) [reply docs/dev/cypress.rst]: Good idea, done!

**Review by pal03377 — DISMISSED** [review]
The additional documentation since my last review is nice! Thanks 👍

File: docs/dev/cypress.rst [hunk 3]
```diff
@@ -18,6 +18,206 @@ the external services (Jira, Bitbucket and Bamboo) and connect them directly wit
 Therefore, the current setup only dynamically deploys the Artemis server and configures it to connect to
 the prelive system, which is already properly setup in the university data center.
 
+
+Local Cypress Setup
+-------------------
+Sometimes developers need to setup Cypress locally, in order to debug failing E2E tests or write new tests.
+In order to setup cypress locally, we need to follow these steps:
```
Comment (Strohgelaender) [inline docs/dev/cypress.rst:25]: https://grammarist.com/spelling/set-up-vs-setup/
Also you used "set up" twice:
```suggestion
Sometimes developers need to set up Cypress locally, in order to debug failing E2E tests or write new tests.
Follow these steps to create your local cypress instance:
```

File: docs/dev/cypress.rst [hunk 4]
```diff
@@ -18,6 +18,206 @@ the external services (Jira, Bitbucket and Bamboo) and connect them directly wit
...
+      "adminUsername": "artemis_admin",
+      "adminPassword": "artemis_admin",
+      "allowGroupCustomization": true,
+      "studentGroupName": "students",
+      "tutorGroupName": "tutors",
+      "editorGroupName": "editors",
+      "instructorGroupName": "instructors"
+    }
+
+  The ``USERID`` part will be replaced by different user ids. These are set within the ``support/users.ts`` file.
```
Comment (Strohgelaender) [inline docs/dev/cypress.rst:57]: To strengthen that cypress will do this and you don't need to change that manually in the file:
```suggestion
  The ``USERID`` part will be automatically replaced by different user ids. These are set within the ``support/users.ts`` file. 
```

File: docs/dev/cypress.rst [hunk 5]
```diff
@@ -18,6 +18,206 @@ the external services (Jira, Bitbucket and Bamboo) and connect them directly wit
...
+
+It's based on Sorry Cypress a open source and selfhostable alternative to the paid cypress cloud.
+
+The dashboard itself can be access here: https://sorry-cypress.ase.cit.tum.de/
+
+To access it, you need these basic auth credentials (sorry cypress itself does not provide an auth 
+system, so we are forced to use nginx basic auth here). You can find these credentials on our confluence page:
+https://confluence.ase.in.tum.de/display/ArTEMiS/Sorry+Cypress+Dashboard
+
+After that you will see the initial dashboard without any useful information.
```
Comment (Strohgelaender) [inline docs/dev/cypress.rst:111]:
```suggestion
After that you will see the initial dashboard. 
```

File: docs/dev/cypress.rst [hunk 6]
```diff
@@ -18,6 +18,206 @@ the external services (Jira, Bitbucket and Bamboo) and connect them directly wit
...
+    :align: center
+    :alt: Sorry Cypress dashboard
+
+Now you get a list of the last runs. In the top right you can enter your branch name to filter the runs.
+
+  .. figure:: cypress/sorry-cypress-runs.png
+    :align: center
+    :alt: Sorry Cypress last runs
+
+The name of the run consists of the branch name followed by the run # . The last part is MySQL or
```
Comment (Strohgelaender) [inline docs/dev/cypress.rst:125]:
```suggestion
The name of the run consists of the branch name followed by the run number. The last part is MySQL or 
```

File: docs/dev/cypress.rst [hunk 7]
```diff
@@ -18,6 +18,206 @@ the external services (Jira, Bitbucket and Bamboo) and connect them directly wit
...
+    :alt: Sorry Cypress dashboard
+
+Now you get a list of the last runs. In the top right you can enter your branch name to filter the runs.
+
+  .. figure:: cypress/sorry-cypress-runs.png
+    :align: center
+    :alt: Sorry Cypress last runs
+
+The name of the run consists of the branch name followed by the run # . The last part is MySQL or 
+PostgreSQL depending on the run environment. If you are in the MySQL project, you will of course only see the MySQL  runs.
```
Comment (Strohgelaender) [inline docs/dev/cypress.rst:126]:
```suggestion
PostgreSQL depending on the run environment. If you are in the MySQL project, you will of course only see the MySQL runs. 
```

**Review by pal03377 — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**